src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorUtil.groovy

```diff
@@ -91,13 +92,31 @@ class IqPolicyEvaluatorUtil
     if (!isNetworkError || failBuildOnNetworkError) {
       throw e
     }
+    else if (isHttp404Error(e)) {
```

Comment:
I think it would be better if somehow we could implement this _only_ for the new API operation; I don't know if we want _every_ 404 to produce a "possible version mismatch" message since that seems incredibly broad in the general case, at least if it's not too difficult to make it more scoped.

Could we just add this particular check as part of a try-catch for the new operation, delegating to what we already had if it's not a 404?

https://github.com/sonatype/nexus-platform-plugin/blob/39b293a3c0e390091a910feeb6ea63374b213ace/src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorUtil.groovy#L52

Another idea, perhaps a bit hackier, would be to extend the `handleNetworkException` method with a `boolean` that would indicate whether or not the message should mention a possible version mismatch; you could declare the boolean at the start of the method as false, set it to true when you make the `verifyOrCreateApplication` call, and then set it back to false once that particular operation had concluded.

Reply:
>  It should be only for the new call to verifyOrCreateApplication.

Changed